### DISCUSSION TITLE
Build result of Server Component (appDir)

### DISCUSSION THREAD

**Discussion by markusra** [vercel/next.js#45844] [Help] [answered]
Hi! 

I am wondering about a change introduced in v13.1.2 regarding production builds. Here are some example build results for v13.1.1 and v13.1.2 respectively (taken from [this CodeSandbox](https://codesandbox.io/p/sandbox/nextjs-build-1cn5w4?file=%2Fapp%2Fcharacter%2F%5Bid%5D%2Fpage.tsx&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A10%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A10%7D%5D)):

**v13.1.1**
<img width="680" alt="Screenshot 2023-02-13 at 09 01 29" src="https://user-images.githubusercontent.com/5471549/218403959-588412ad-818a-4efd-8b04-1c7c9c628093.png">

**v13.1.2**
<img width="578" alt="Screenshot 2023-02-13 at 09 02 47" src="https://user-images.githubusercontent.com/5471549/218404014-8eaeb799-dcab-48df-b272-93eee156efd1.png">

As you can see, the dynamic route `/character/[id]` was server-side rendered at runtime in earlier versions. In the v13.1.2 build, the same route is statically rendered. The route uses the new data fetching system inside Server Components where the dynamic param `id` is used to fetch specific data.

The problem with this is that the fetch is actually triggered with the string `[id]` (i.e. `/character/%5Bid%5D`) at build-time which obviously results in an error from the API. This in turn can fail the build.

Does anybody know the background for this change or have any tips on how to adopt?

**Comment by icyJoseph** [comment, marked as answer]
Probably a regression.

I tried on `latest` and it behaves as you describe for `13.1.1` - beta phases might include bugs and what not... On 13.1.2 it doesn't even listen to segment config, https://beta.nextjs.org/docs/api-reference/segment-config#dynamic

I'd say, take latest if possible.

## Edit

I had left `export const dynamic = 'force-dynamic'` on my page... 

```tsx
import { PokeCard } from "components/PokeCard";
import { fetchPokemon } from "lib/pokemon";

// const initial =  [1, 4, 7];

// export const dynamicParams = true;

// export async function generateStaticParams() {
//   return initial.map((id) => ({
//     id: `${id}`,
//   }));
// }

export const dynamic = "force-dynamic";

async function PokeView({ params }: { params: { id: string } }) {
  console.log(params);
  const pokemon = await fetchPokemon(params.id);

  return <PokeCard pokemon={pokemon} />;
}

export default PokeView;
```

This variation works too:

```tsx
import { PokeCard } from "components/PokeCard";
import { fetchPokemon } from "lib/pokemon";

const initial = [1, 4, 7];

export const dynamicParams = true;

export async function generateStaticParams() {
  return initial.map((id) => ({
    id: `${id}`,
  }));
}

// export const dynamic = "force-dynamic";

async function PokeView({ params }: { params: { id: string } }) {
  console.log(params);
  const pokemon = await fetchPokemon(params.id);

  return <PokeCard pokemon={pokemon} />;
}

export default PokeView;
```

**Reply by markusra** [reply]
Thank you! Tried setting `export const dynamic = "force-dynamic"` earlier, but it had no effect in v13.1.2 as you pointed out. Upgrading to `latest` did the trick!
